instrumentation/kafka/kafka-streams-0.11/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/kafkastreams/KafkaStreamReflectionUtil.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.kafkastreams;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Properties;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.kstream.KStream;
+
+/**
+ * kafka stream reflection util which is used to compatible with different versions of kafka stream
+ */
+class KafkaStreamReflectionUtil {
+
+  private KafkaStreamReflectionUtil() {}
+
+  @SuppressWarnings("ClassNewInstance")
+  static Object createBuilder()
+      throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+    try {
+      // Different class names for test and latestDepTest.
+      return Class.forName("org.apache.kafka.streams.kstream.KStreamBuilder").newInstance();
+    } catch (ClassNotFoundException
+        | NoClassDefFoundError
+        | InstantiationException
+        | IllegalAccessException e) {
+      return Class.forName("org.apache.kafka.streams.StreamsBuilder").newInstance();
+    }
+  }
```

Comment:
perhaps we could avoid the need for the  `@SuppressWarnings("ClassNewInstance")` if we did something like this?

```suggestion
  static Object createBuilder() throws Exception {
    try {
      // Different class names for test and latestDepTest.
      return Class.forName("org.apache.kafka.streams.kstream.KStreamBuilder").getConstructors()[0].newInstance();
    } catch (Exception e) {
      return Class.forName("org.apache.kafka.streams.StreamsBuilder").getConstructors()[0].newInstance();
    }
  }
```